lib/maths/CDataFrameCategoryEncoder.cc

```diff
@@ -24,6 +25,37 @@ using TDoubleVec = std::vector<double>;
 using TSizeDoublePr = std::pair<std::size_t, double>;
 using TSizeDoublePrVec = std::vector<TSizeDoublePr>;
 using TSizeDoublePrVecVec = std::vector<TSizeDoublePrVec>;
+using TSizeUSet = boost::unordered_set<std::size_t>;
+
+const std::size_t METRIC_CATEGORY{std::numeric_limits<std::size_t>::max()};
+const std::size_t RARE_CATEGORY{METRIC_CATEGORY - 1};
+const std::size_t TARGET_MEAN_CATEGORY{RARE_CATEGORY - 1};
+const std::size_t TARGET_CATEGORY{TARGET_MEAN_CATEGORY - 1};
```

Comment:
I'm not sure this is worth it: we're never going to run into this case in practice because we limit the distinct categories to be the largest integer representable by a float. So it just adds unnecessary complexity to the code.